### PR TITLE
Ruby: expand explanation of desugaring

### PR DESCRIPTION
**Pull request by aibaars**

### CODE REVIEW

File: docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst [hunk 1]
```diff
@@ -421,7 +421,13 @@ All classes in this subsection are subclasses of VariableAccess_.
 Desugaring
 ~~~~~~~~~~
 
-Certain Ruby language features are implemented using syntactic sugar. For example, supposing that ``x`` is an object with an attribute ``foo``, the assignment::
+Certain Ruby language features are shorthands for a common operations that could also be expressed in an alternate, more verbose, forms.
```
Comment (nickrolfe) [inline docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst:424]:
```suggestion
Certain Ruby language features are shorthands for common operations that could also be expressed in other, more verbose, forms.
```

File: docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst [hunk 2]
```diff
@@ -421,7 +421,13 @@ All classes in this subsection are subclasses of VariableAccess_.
 Desugaring
 ~~~~~~~~~~
```
Comment (felicitymay) [inline docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst:422]: How about:

```suggestion
Syntactic sugar and desugaring
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst [hunk 3]
```diff
@@ -421,7 +421,13 @@ All classes in this subsection are subclasses of VariableAccess_.
...
 
-Certain Ruby language features are implemented using syntactic sugar. For example, supposing that ``x`` is an object with an attribute ``foo``, the assignment::
+Certain Ruby language features are shorthands for a common operations that could also be expressed in an alternate, more verbose, forms.
+Such language features are typically referred to as "syntactic sugar", and make it easier for programmers to write and read code. This is
+great for programmers. Source code analyzers on the other hand, this lead to additional work as they need to understand the short 
+hand notation as well as the long form. To make analysis easier, CodeQL automatically "desugars" Ruby code, effectively rewriting
```
Comment (nickrolfe) [inline docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst:427]:
```suggestion
great for programmers. For source code analyzers, however, this leads to additional work as they need to understand the short-hand notation as well as the long form. To make analysis easier, CodeQL automatically "desugars" Ruby code, effectively rewriting
```

**Review by nickrolfe — COMMENTED** [review]
I'm not a copy-writer, but I spotted a couple of grammatical fixes and simplifications.

File: docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst [hunk 4]
```diff
@@ -421,7 +421,12 @@ All classes in this subsection are subclasses of VariableAccess_.
 Desugaring
 ~~~~~~~~~~
 
-Certain Ruby language features are implemented using syntactic sugar. For example, supposing that ``x`` is an object with an attribute ``foo``, the assignment::
+Certain Ruby language features are shorthands for common operations that could also be expressed in other, more verbose, forms.
+Such language features are typically referred to as "syntactic sugar", and make it easier for programmers to write and read code. This is
+great for programmers. For source code analyzers, however, this leads to additional work as they need to understand the short-hand notation as well as the long form. To make analysis easier, CodeQL automatically "desugars" Ruby code, effectively rewriting
```
Comment (felicitymay) [inline docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst:426]:
```suggestion
great for programmers. For source code analyzers, however, this leads to additional work as they need to understand the shorthand notation as well as the long form. To make analysis easier, CodeQL automatically "desugars" Ruby code, effectively rewriting
```

File: docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst [hunk 5]
```diff
@@ -421,7 +421,12 @@ All classes in this subsection are subclasses of VariableAccess_.
...
+rich syntactic constructs into equivalent code that uses simpler syntactic contructs.
```
Comment (felicitymay) [inline docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst:427]:
```suggestion
rich syntactic constructs into equivalent code that uses simpler syntactic constructs.
```

File: docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst [hunk 6]
```diff
@@ -432,7 +437,7 @@ is desugared to code similar to::
 
 In other words, there is effectively a call to the SetterMethodCall_ ``foo=`` on ``x`` with argument ``__synth_0 = y``, followed by a read of the ``__synth_0`` variable.
 
-In CodeQL, this is implemented by syntheisizing AstNode_ instances corresponding to this desugared version of the code.
+In CodeQL, this is implemented by synthesizing AstNode_ instances corresponding to this desugared version of the code.
 
 Note that both the original AssignExpr_ and the desugared SetterMethodCall_ versions are both available to CodeQL queries, and it is usually not necessary to be aware of any desugaring that may take place. However, if a codebase explicitly uses ``x.foo=(y)`` SetterMethodCall_ syntax, then this will not be found by a query for AssignExpr_ instances.
```
Comment (felicitymay) [inline docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst:442]: Two boths, suggest:

```suggestion
Note that the original AssignExpr_ and the desugared SetterMethodCall_ versions are both available to use in CodeQL queries, and you do not usually need to be aware of any desugaring that may take place. However, if a codebase explicitly uses ``x.foo=(y)`` SetterMethodCall_ syntax, you cannot find this syntax by searching for instances of AssignExpr_ .
```

**Review by felicitymay — COMMENTED** [review]
This was updated while I was reviewing it. I've included a few more suggestions.

**Review by felicitymay — APPROVED** [review]
Thanks for the updates to this section. This seems clearer to me 👍🏻 

I'm happy for you to merge when the tests complete - especially the Docs - Generate Sphinx test 🙂

**Pull request merged**